### PR TITLE
Proposal (mainly )for issues #4 and #7

### PR DESCRIPTION
**Pull request by mgiulio**
[Description](https://github.com/mgiulio/barcelonajs.github.io/wiki)

There is still some work to do. 
This is just a personal interpretation of the site, just to get the ball rolling.

### CODE REVIEW

**Comment by mgiulio** [comment]
Ok @PatrickHeneise , feel free to review, test and merge this PR.

The talk page is not RWD yet.

Along the way I fixed something in the build process. These are the relative commits:

* https://github.com/BarcelonaJS/barcelonajs.github.io/pull/16/commits/2f89d1f1da0022691303540ae8915a1bcff1e0f5
* https://github.com/BarcelonaJS/barcelonajs.github.io/pull/16/commits/c6266ab45520ac1a36ed57bb9668ef7e85323c12
* https://github.com/BarcelonaJS/barcelonajs.github.io/pull/16/commits/43f70d8f0cea8189c8688e6b89e59c74b7a61ecb
* https://github.com/BarcelonaJS/barcelonajs.github.io/pull/16/commits/4caddde2597a09b0286912340465f149b6abad80

**Comment by eth0lo** [comment]
I just check it visually, it looks ok.

I need to deepdive into the code itself, but one thing that I notice it's that you change the multiple declarations for only one; personally I would like you to keep the style that was previously.

File: build.js [hunk 1]
```diff
@@ -31,43 +40,156 @@ function normalize (options) {
   return options
 }
 
-var bcnjs = function bcnjs (opts) {
+function bcnjs(opts) {
```
Comment (eth0lo) [inline build.js]: This middleware is huge, can you make it smaller?

File: build.js [hunk 2]
```diff
@@ -83,29 +205,52 @@ Metalsmith(__dirname)
...
   .use(markdown())
   .use(layouts({
     engine: 'handlebars',
     directory: 'src/layouts',
     partials: 'src/partials'
   }))
+  .use(bumpTalkUrl())
+  .use(permalinks({
+    pattern: ':title'
+  ,relative: false
```
Comment (eth0lo) [inline build.js]: commas at the end pls.

Really keep the previous style

File: src/partials/socialButtons.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="social-buttons">
+	{{> twitterShareBtn urlToShare = urlToShare text = name}}
+	{{> facebookShareBtn urlToShare = urlToShare}}
+	{{> googleplusShareBtn urlToShare = urlToShare}}
+	{{> linkedinShareBtn urlToShare = urlToShare title = name}}
+</div>
```
Comment (eth0lo) [inline src/partials/socialButtons.html]: New line

File: src/partials/talk/main.html [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<div class="main">
+	<h1 class="title">{{name}}</h1>
+	<div class="description">{{{contents}}}</div>
+	<div class="material">
+		<h2 class="title">Material</h2>
+		{{> talk/material}}
+	</div>
+</div>
```
Comment (eth0lo) [inline src/partials/talk/main.html]: New line

File: src/partials/talk/speaker.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+	<h3 class="name">{{name}}</h3>
+	<ul class="socials">
+		{{#if twitter}}
+			<li>{{> twitterLink username = twitter}}</li>
+		{{/if}}
+		{{#if github}}
+			<li>{{> githubLink username = github}}</li>
+		{{/if}}
+	</ul>
+</div>
```
Comment (eth0lo) [inline src/partials/talk/speaker.html]: New line

File: src/partials/talk/sponsorsGrid.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<ul class="sponsors-grid">
+	{{#each sponsors}}
```
Comment (eth0lo) [inline src/partials/talk/sponsorsGrid.html]: Looks like this is a tab, we were using spaces

File: src/partials/talk/talkinfo.html [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+		</li>
+	</ul>
+	{{#if tags}}
+		<ul class="tags">
+			{{#each tags}}
+				<li>{{this}}</li>
+			{{/each}}
+		</ul>
+	{{/if}}
+</div>
```
Comment (eth0lo) [inline src/partials/talk/talkinfo.html]: New line

File: src/sass/barcelonajs.scss [hunk 8]
```diff
@@ -145,7 +148,7 @@ section {
     }
     .talk-title {
       color: $dark;
-      font-size: 1rem;
+	  line-height: 1.4;
```
Comment (eth0lo) [inline src/sass/barcelonajs.scss]: Spaces

File: src/sass/barcelonajs.scss [hunk 9]
```diff
@@ -254,4 +257,11 @@ input[type="email"] {
   .talk-submit-form__input-wrapper {
     margin-right: 12px;
   }
-}
\ No newline at end of file
+}
+
+.page-wrap {
+	max-width: 1200px;
```
Comment (eth0lo) [inline src/sass/barcelonajs.scss]: Spaces

File: src/sass/talk/index.scss [hunk 10]
```diff
@@ -0,0 +1,208 @@
+body.talk {
+	background-color: #f5f2f2;
+}
+
+@import 'layout';
+
+.talk .page-wrap .main > .title {
```
Comment (eth0lo) [inline src/sass/talk/index.scss]: Try to use less specifity in the selectors

File: src/sass/talk/layout.scss [hunk 11]
```diff
@@ -0,0 +1,55 @@
+$sidebar-width: 300px;
+$secondary-max-height: 620px;
+$two-col-breakpoint: 800px;
+$three-col-breakpoint: 1100px;
+
+.talk {
+	.page-wrap {
+		&::after {
+			content: '';
```
Comment (eth0lo) [inline src/sass/talk/layout.scss]: Why a clearfix when two displays inline block would be enough?

**Comment by mgiulio** [comment]
Hi @eth0lo , thanks for the feedback!

Just had a quick glance at your comments. Not probs for multiple decls, spaces, tabs, newlines and other style-preserving changes. 
At the moment I'm a bit busy, I'll be back here ASAP.

Cheers, mg

**Comment by eth0lo** [comment]
Also pls check the bcnjs middleware is kind of big it would be awesome, if it could be splitted in several smaller ones

**Comment by PatrickHeneise** [comment]
we can just add `standard` as a requirement, then the style issues won't be a problem / easy fix/

**Comment by mgiulio** [comment]
@eth0lo Ok for the middleware split.

**Comment by mgiulio** [comment]
@PatrickHeneise Are you talking about https://github.com/feross/standard ?

Anyway, I knew I didn't 100% respected the src style. I planned to fix this in a later phase(that is, now :) )

**Comment by PatrickHeneise** [comment]
Yeah, standardjs is pretty cool when it comes to code style :)

**Pull request merged**